modules/vsphere-problem-detector-config-checks.adoc

```diff
@@ -0,0 +1,82 @@
+// Module included in the following assemblies:
+//
+// * installing/installing_vsphere/using-vsphere-problem-detector-operator.adoc
+
+:operator-name: vSphere Problem Detector Operator
+
+[id="vsphere-problem-detector-config-checks_{context}"]
+= Configuration checks run by the vSphere Problem Detector Operator
+
+The following tables identify the configuration checks that are run by the {operator-name}. Some checks verify the configuration of the cluster. Other checks verify the configuration of each node in the cluster.
+
+.Cluster configuration checks
+[options="header",cols="20,80a"]
+|===
+|Name
+|Description
+
+|`CheckDefaultDatastore`
+|Verifies that the default datastore name in the vSphere configuration is short enough to be used with dynamic provisioning.
+
+If this check fails, you can expect the following:
+
+* `systemd` logs errors to the journal such as `Failed to set up mount unit: Invalid argument`.
+
+* `systemd` does not unmount volumes if the virtual machine is shut down or rebooted without draining all the pods from the node.
+
+If this check fails, reconfigure vSphere with a shorter name for the default datastore.
+
+|`CheckFolderPermissions`
+|Verifies the permission to list volumes in the default datastore. This permission is required to create volumes. The Operator verifies the permission by listing the `/` and `/kubevols` directories. The root directory must exist. It is acceptable if the `/kubevols` directory does not exist when the check runs. The `/kubevols` directory is created when the datastore is used with dynamic provisioning if the directory does not already exist.
+
+If this check fails, review the required permissions for the vCenter account that was specified during the {product-title} installation.
+
+|`CheckPVs`
+|Verifies that the fully qualified path to each persistent volume is less than 255 characters.
+
+If this check fails, rename the VDMK file to use a shorter name.
+
+|`CheckStorageClasses`
+|Verifies the following:
```

Comment:
Would it be better to join `CheckPVs` and `CheckStorageClasses` with `CheckDefaultDatastore` above? They basically check the same thing, the datastore name used in the default config, in PVs or in StorageClasses must be short enough. The consequences are well described in `CheckDefaultDatastore`.

It's hard to tell what is "short enough" - the datastore name gets escaped by systemd (e.g. each `-` yields into 4 characters!) and it gets used in a path `/var/lib/kubelet/plugins/kubernetes.io/vsphere-volume/mounts/<datastore name> <uuid>/<cluster-name>-dynamic-pvc-<uuid>` and this path must be under 255 characters. All other components of the path are set, so users can affect only `<datastore name>` there.

Reply:
My preference at this point is to leave them as-is.  They are separate checks and if a customer can fail one check without failing the others, then the customer might be unsure of which remedy to attempt.

Reply:
Ack